### ISSUE TITLE
failed generations show up as flagged by safety checker

### ISSUE THREAD

**Issue by iFrostizz** [brycedrennan/imaginAIry#83]
Hello,

First I wanted to thank you for the simplicity of the setup of this stable diffusion CLI.

Just tried it with a Nvidia GTX 1660 super with 6GB of RAM, but any prompt will output a red-ish image:

```sh
$ imagine "a photo of a dog"
```

![000000_485200245_plms40_PS7 5_a_scenic_landscape_ generated](https://user-images.githubusercontent.com/51274081/200933681-5bbad2b7-e8ca-4bc8-b546-b675a0ec7f0d.jpg)

edit: it may somehow be related to: https://github.com/Sygil-Dev/sygil-webui/issues/34

Let me know if there is any details I could provide to help you :)

Thanks !

**Comment by brycedrennan** [comment]
The red color implies it's the NSFW filter. But the fact that it's doing that for every image implies something is probably wrong with the generation. The NSFW filter tends to just flag any blurry image. You could maybe try some different samplers and see if that helps.

**Comment by brycedrennan** [comment]
Possibly related: https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/4558

**Comment by brycedrennan** [comment]
Can you try this again with 5.0.0 which was just released? If it's still broken can you run it with `--show-work` and see what's going on?

**Comment by iFrostizz** [comment]
Still doesn't work :/ the `--show-work` is interesting though !
Wanted to wrap the images into a `summary` tag in order to make it collapsible, but the images wouldn't render, so I'm sorry for this.

![00000001_S72991062_0000_step000_neutral_conditioning](https://user-images.githubusercontent.com/51274081/201543072-dae614ae-e48d-45a8-924a-bf8ed0e11a17.jpg)
![00000001_S72991062_0000_step000_positive_conditioning](https://user-images.githubusercontent.com/51274081/201543076-b4f96ed2-a3dc-467a-a242-341013d74bd8.jpg)
![00000001_S72991062_0001_step000_initial_latent-torch Size_ 1,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543077-5b92189e-0d5c-423a-a078-541293e39d32.jpg)
![00000001_S72991062_0002_step000_initial_sigma_noised_tensor-torch Size_](https://user-images.githubusercontent.com/51274081/201543078-2b55f44e-b27d-4f20-ba43-5c18b7f50be8.jpg)
![00000001_S72991062_0003_step000_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543081-1417dea7-2e7e-4894-91fb-9c8e9e4eae23.jpg)
![00000001_S72991062_0004_step000_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543083-4fc428a7-72ce-4175-9556-f156f08de713.jpg)
![00000001_S72991062_0005_step000_noisy_latent-torch Size_ 1,_4,_64,_64 _](https://user-images.githubusercontent.com/51274081/201543085-f2bad2b6-f164-45f8-80d7-5ef3857cbd2d.jpg)
![00000001_S72991062_0006_step000_predicted_latent-torch Size_ 1,_4,_64,_6](https://user-images.githubusercontent.com/51274081/201543086-7f70d90d-e3b9-45f5-9ca3-6a543192d6f1.jpg)
![00000001_S72991062_0007_step001_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543087-036e3fb4-f4ee-499e-8e56-ab1f798b0850.jpg)
![00000001_S72991062_0008_step001_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543088-ac8a5b4e-3353-4cf1-ad90-6e0179483d13.jpg)
![00000001_S72991062_0009_step001_noisy_latent-torch Size_ 1,_4,_64,_64 _](https://user-images.githubusercontent.com/51274081/201543089-8fd5cf47-51c7-4e63-89b3-769dd50be97c.jpg)
![00000001_S72991062_0010_step001_predicted_latent-torch Size_ 1,_4,_64,_6](https://user-images.githubusercontent.com/51274081/201543090-4d80a2a2-364c-490e-ba0b-eccb13504b45.jpg)
![00000001_S72991062_0011_step002_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543091-18db67d2-dbba-47f8-91bd-6985a4eadb54.jpg)
![00000001_S72991062_0012_step002_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543092-9f58ccdf-4c00-4c58-b88d-e19c4df7ab79.jpg)
![00000001_S72991062_0013_step002_noisy_latent-torch Size_ 1,_4,_64,_64 _](https://user-images.githubusercontent.com/51274081/201543093-2d7fd346-7662-4618-847b-3a22c46b31f3.jpg)
![00000001_S72991062_0014_step002_predicted_latent-torch Size_ 1,_4,_64,_6](https://user-images.githubusercontent.com/51274081/201543095-c3dd91b6-f4a4-4e06-a5b2-1b94352f5a27.jpg)
![00000001_S72991062_0015_step003_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543096-5f13de97-4176-49d4-ac7c-03edfab5e0c0.jpg)
![00000001_S72991062_0016_step003_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543097-74bdb45a-04cf-4874-b1ba-d05d26c3f0ac.jpg)
![00000001_S72991062_0017_step003_noisy_latent-torch Size_ 1,_4,_64,_64 _](https://user-images.githubusercontent.com/51274081/201543098-08397c9d-67f1-4963-a7a4-120cafa7436b.jpg)
![00000001_S72991062_0018_step003_predicted_latent-torch Size_ 1,_4,_64,_6](https://user-images.githubusercontent.com/51274081/201543099-70ef8916-e8d1-4b53-9e9a-1e6c5003d866.jpg)
![00000001_S72991062_0019_step004_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543101-a05776c9-51c0-4883-8a0e-72d8ed93640e.jpg)
![00000001_S72991062_0020_step004_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543102-b4ce03a7-5c79-47e9-b477-05172533ab8c.jpg)
![00000001_S72991062_0021_step004_noisy_latent-torch Size_ 1,_4,_64,_64 _](https://user-images.githubusercontent.com/51274081/201543103-bd385a84-7b6c-4078-9b14-447fc48e0d12.jpg)
![00000001_S72991062_0022_step004_predicted_latent-torch Size_ 1,_4,_64,_6](https://user-images.githubusercontent.com/51274081/201543105-64b6cb9d-1e9b-4a42-8406-7832947ecd4c.jpg)
![00000001_S72991062_0023_step005_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543106-effd6258-7710-42a2-b18d-7ecf34b1790f.jpg)
![00000001_S72991062_0024_step005_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543108-ad33b75d-7c22-4773-8d9f-798a063b1a89.jpg)
![00000001_S72991062_0025_step005_noisy_latent-torch Size_ 1,_4,_64,_64 _](https://user-images.githubusercontent.com/51274081/201543109-ca0540fc-ab29-4478-9b06-b1feaa6e45a2.jpg)
![00000001_S72991062_0026_step005_predicted_latent-torch Size_ 1,_4,_64,_6](https://user-images.githubusercontent.com/51274081/201543110-a78743b5-7437-4938-9dfa-42474282a8aa.jpg)
![00000001_S72991062_0027_step006_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543111-dfbdffdb-a119-4946-8613-9e9320a04473.jpg)
![00000001_S72991062_0028_step006_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543112-d3498903-76e7-45c8-a5da-b79f96e18cab.jpg)
![00000001_S72991062_0029_step006_noisy_latent-torch Size_ 1,_4,_64,_64 _](https://user-images.githubusercontent.com/51274081/201543114-7f508054-67d4-4680-902e-40a2f45911c0.jpg)
![00000001_S72991062_0030_step006_predicted_latent-torch Size_ 1,_4,_64,_6](https://user-images.githubusercontent.com/51274081/201543115-3ff459ce-6cfb-46b0-88aa-d0a859d1e54d.jpg)
![00000001_S72991062_0031_step007_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543116-d2d8d330-1b0f-4c74-b79c-79c59c82aa04.jpg)
![00000001_S72991062_0032_step007_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543117-3b980172-e637-4618-88ce-4d8c4b692319.jpg)
![00000001_S72991062_0033_step007_noisy_latent-torch Size_ 1,_4,_64,_64 _](https://user-images.githubusercontent.com/51274081/201543118-ecc0191c-5601-4440-9499-38ef6b9a8196.jpg)
![00000001_S72991062_0034_step007_predicted_latent-torch Size_ 1,_4,_64,_6](https://user-images.githubusercontent.com/51274081/201543119-47605dfa-acec-4768-b203-9346aeaa9151.jpg)
![00000001_S72991062_0035_step008_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543121-ce73a868-0d93-4976-9907-e08fab93cf48.jpg)
![00000001_S72991062_0036_step008_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543122-2ebd37ca-b6a0-4fe7-91d2-ea7b8e1b6717.jpg)
![00000001_S72991062_0037_step008_noisy_latent-torch Size_ 1,_4,_64,_64 _](https://user-images.githubusercontent.com/51274081/201543123-9597749f-3369-45fb-82d4-3daf424eba65.jpg)
![00000001_S72991062_0038_step008_predicted_latent-torch Size_ 1,_4,_64,_6](https://user-images.githubusercontent.com/51274081/201543124-c8e77b15-4fe3-47da-9097-f7bab3b753a2.jpg)
![00000001_S72991062_0039_step009_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543125-8fe3e680-ee9c-4ac5-8d85-cf39a6d86b37.jpg)
![00000001_S72991062_0040_step009_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543126-c18f7aa7-f311-45c9-ad60-a5c99960e322.jpg)
![00000001_S72991062_0041_step009_noisy_latent-torch Size_ 1,_4,_64,_64 _](https://user-images.githubusercontent.com/51274081/201543127-99dc62de-570c-43b5-99b7-57526e47db44.jpg)
![00000001_S72991062_0042_step009_predicted_latent-torch Size_ 1,_4,_64,_6](https://user-images.githubusercontent.com/51274081/201543128-9ac29d7c-318b-47c4-8cd7-d460babbca69.jpg)
![00000001_S72991062_0043_step010_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543129-7610542f-5cf5-494c-9f65-11d89f9deda9.jpg)
![00000001_S72991062_0044_step010_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543131-806892cf-0c78-49ea-bd50-abfeec072191.jpg)
![00000001_S72991062_0045_step010_noisy_latent-torch Size_ 1,_4,_64,_64 _](https://user-images.githubusercontent.com/51274081/201543132-e9bf41a9-ed6c-4e8e-84ba-61a13bb1a718.jpg)
![00000001_S72991062_0046_step010_predicted_latent-torch Size_ 1,_4,_64,_6](https://user-images.githubusercontent.com/51274081/201543134-21a99db2-4717-4cc5-8578-c3de63c905eb.jpg)
![00000001_S72991062_0047_step011_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543135-a5ef0f8b-f8af-4047-9c57-fcfd0aed9af2.jpg)
![00000001_S72991062_0048_step011_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543137-fb12af1a-9298-4832-8e8e-a6a94f23cb60.jpg)
![00000001_S72991062_0049_step011_noisy_latent-torch Size_ 1,_4,_64,_64 _](https://user-images.githubusercontent.com/51274081/201543138-155c9335-9b0c-448c-b51d-71269462341f.jpg)
![00000001_S72991062_0050_step011_predicted_latent-torch Size_ 1,_4,_64,_6](https://user-images.githubusercontent.com/51274081/201543140-6e251fec-24b6-4d48-88c8-95613b9fbf70.jpg)
![00000001_S72991062_0051_step012_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543141-f851ad71-ffbb-4df9-8527-5dc915250ee9.jpg)
![00000001_S72991062_0052_step012_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543143-53e2e603-e0d7-49fc-8e5c-423b7e4455e1.jpg)
![00000001_S72991062_0053_step012_noisy_latent-torch Size_ 1,_4,_64,_64 _](https://user-images.githubusercontent.com/51274081/201543144-d212d7b5-afe3-4343-8f1e-c3e079372f1a.jpg)
![00000001_S72991062_0054_step012_predicted_latent-torch Size_ 1,_4,_64,_6](https://user-images.githubusercontent.com/51274081/201543145-a18c8235-0d09-426d-87f0-b41849537aae.jpg)
![00000001_S72991062_0055_step013_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543146-a54fc745-99ed-4daa-934a-638a0e320708.jpg)
![00000001_S72991062_0056_step013_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543148-09934106-7ce0-417e-9ef0-fd183a098f26.jpg)
![00000001_S72991062_0057_step013_noisy_latent-torch Size_ 1,_4,_64,_64 _](https://user-images.githubusercontent.com/51274081/201543149-528d5c7f-6a2e-441a-9e88-bde5ac31e086.jpg)
![00000001_S72991062_0058_step013_predicted_latent-torch Size_ 1,_4,_64,_6](https://user-images.githubusercontent.com/51274081/201543153-0101c2dd-f747-4096-a9a9-1f666000ca42.jpg)
![00000001_S72991062_0059_step014_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543155-70a45fbe-5ab4-4248-93c8-677f9af992fe.jpg)
![00000001_S72991062_0060_step014_predicted_noise-torch Size_ 2,_4,_64,_64](https://user-images.githubusercontent.com/51274081/201543159-d5152e80-59e0-4580-8686-d08430a1fc4d.jpg)
![00000001_S72991062_0061_step014_noisy_latent-torch Size_ 1,_4,_64,_64 _](https://user-images.githubusercontent.com/51274081/201543160-95d160ba-3539-47fd-84d2-0c5a47794969.jpg)
![00000001_S72991062_0062_step014_predicted_latent-torch Size_ 1,_4,_64,_6](https://user-images.githubusercontent.com/51274081/201543161-dd956b2f-788b-4b89-bfd4-68310c66ef53.jpg)

**Comment by brycedrennan** [comment]
Thanks.  Yeah something is broken for sure.  Unfortunately without your graphics card it will be hard for me to debug.

This issue also seems similar. It wasn't obvious to me from a quick skim how they fixed it. 
https://github.com/invoke-ai/InvokeAI/issues/526 

Maybe you'll have better luck with that version of stable diffusion?

**Comment by brycedrennan** [comment]
I'm inclined to just say imaginairy doesn't support this but if someone want's to figure out the problem I'd be happy to merge a fix.

**Comment by AntouanK** [comment]
I'm having similar issues when I try to make masks of my face.
Can we have a hint on how to disable nsfw so we can see what the output really is?

I tried changing some files in `~/.local/lib/python3.10/site-packages/diffusers/pipelines/stable_diffusion/` but the result is always the same. ( not a python dev, so probably I do something wrong )

**Comment by brycedrennan** [comment]
Only when you do masks? what version of imaginairy? what OS? what graphics card?

**Comment by AntouanK** [comment]
Yeah, on some photos it's not cropping the face correctly, so I did `--steps 1` and I started playing with the mask prompt.
But on some photos I get the nsfw ( although there's absolutely nothing nsfw on the photo )

Linux / Arch
GTX 1070 8GB

**Comment by brycedrennan** [comment]
Yeah I imagine that's kind of annoying.  The filter basically always triggers if you don't have enough steps. 

This library doesn't use the diffusers stable diffusion. You'd need to edit things in `imaginairy`. 

Might be a stretch but I'd just directly call `get_img_mask` in python and save the mask to disk to iterate quickly.
https://github.com/brycedrennan/imaginAIry/blob/master/imaginairy/enhancers/clip_masking.py#L34

**Comment by AntouanK** [comment]
no idea how to do that :D 
I'll look up some python tutorials to see how to call it.

thank you!

**Comment by djpavlo** [comment]
the same on the mac M1 processor.

**Comment by brycedrennan** [comment]
@djpavlo this thread is about an issue that happens on CUDA GPUs.  Please open a new issue with a detailed description of what's happening, what python version, and a copy/paste of the terminal output you're getting.

**Comment by reubenfirmin** [comment]
I had the same issue far too often, and solved it by hacking SafetyFilter to return False on both functions, and then initialized a new SafetyFilter rather than calling the safety model. You are right that the images in question are blurry, but it helped me figure out what was happening.

While making that change I saw the environment variable you have for relaxing the filter at least. I'd suggest providing a --nsfw=off|relaxed|strict param; strict can be default, but anybody who wants to can do what I did, so it's more user friendly to provide that as a param.
